server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/delivery/SieveMailet.java

```diff
@@ -58,44 +89,111 @@ public SieveMailet(UsersRepository usersRepos, MailboxManager mailboxManager, Re
     }
 
     @Override
-    public void init(MailetConfig config) throws MessagingException {
+    public void init() throws MessagingException {
+
+        this.deliveryHeader = getInitParameter("addDeliveryHeader");
+        this.resetReturnPath = getInitParameter("resetReturnPath", true);
+        this.consume = getInitParameter("consume", true);
+        this.verbose = getInitParameter("verbose", false);
+        this.quiet = getInitParameter("quiet", false);
+
+        actionDispatcher = new ActionDispatcher();
+
         setLocator(resourceLocator);
         setPoster(this);
-        super.init(config);
+
+        if (poster == null || locator == null) {
+            throw new MailetException("Not initialised. Please ensure that the mailet container supports either" +
+                " setter or constructor injection");
+        }
+
+        try {
+            final ConfigurationManager configurationManager = new ConfigurationManager();
+            final int logLevel;
+            if (verbose) {
+                logLevel = CommonsLoggingAdapter.TRACE;
+            } else if (quiet) {
+                logLevel = CommonsLoggingAdapter.FATAL;
+            } else {
+                logLevel = CommonsLoggingAdapter.WARN;
+            }
+            log = new CommonsLoggingAdapter(this, logLevel);
+            configurationManager.setLog(log);
+            factory = configurationManager.build();
+        } catch (SieveConfigurationException e) {
+            throw new MessagingException("Failed to load standard Sieve configuration.", e);
+        }
+    }
+
+    public ResourceLocator getLocator() {
+        return locator;
+    }
+
+    public void setLocator(ResourceLocator locator) {
+        this.locator = locator;
+    }
+
+    public Poster getPoster() {
+        return poster;
+    }
+
+    public void setPoster(Poster poster) {
+        this.poster = poster;
+    }
+
+    public boolean isConsume() {
+        return consume;
+    }
+
+    public void setConsume(boolean consume) {
+        this.consume = consume;
+    }
+
+    public boolean isVerbose() {
+        return verbose;
+    }
+
+    public void setVerbose(boolean verbose) {
+        this.verbose = verbose;
+    }
+
+    public boolean isQuiet() {
+        return quiet;
+    }
+
+    public void setQuiet(boolean quiet) {
+        this.quiet = quiet;
+    }
+
+    public boolean isInfoLoggingOn() {
+        return verbose || !quiet;
     }
 
-    /**
-     * Return the username to use for sieve processing for the given
-     * MailAddress. If virtualhosting is supported use the full emailaddrees as
-     * username
-     * 
-     * @param m
-     * @return username
-     */
     protected String getUsername(MailAddress m) {
         try {
             if (usersRepos.supportVirtualHosting()) {
                 return m.toString();
             } else {
-                return super.getUsername(m);
+                return m.getLocalPart() + "@localhost";
```

Comment:
It's old code, simply moved.

To answer your question : they need username to have a domain to ease parsing.

It turns out, it's part of JSieve implementation, so out of the scope of this refactoring ;-) (See later commits, I make it optional)